src/component/processes/end.ts

```diff
@@ -5,51 +5,57 @@ import { itemAdapterEmpty } from '../utils/adapter';
 export default class End {
 
   static run(scroller: Scroller, process: Process, payload: any = {}) {
+    const { workflow, state } = scroller;
     const { error } = payload;
 
-    scroller.state.workflowOptions.reset();
+    state.workflowOptions.reset();
 
     if (!error && process !== Process.reload) {
       // set out params, accessible via Adapter
       End.calculateParams(scroller);
     }
 
+    // explicit interruption for we don't want go through the workflow loop finalizing
+    if ((<any>workflow.call).interrupted) {
+      return workflow.call();
+    }
+
     // what next? done?
     const next = End.getNext(scroller, process, error);
 
     // need to apply Buffer.items changes if clip
-    if (scroller.state.clip.doClip) {
+    if (state.clip.doClip) {
```

Comment:
can this have some sort of 'force' on it? Or where does the doClip get set to true? In my case, I'm calling reload because my data has changed. Meaning that I will always want to run change detection after.

Reply:
@lmreicks Angular Change detector is being run by the scroller twice: 

 - on new items render (asynchronously)
 - on old items clip (synchronously), this one you mentioned above

The Reload process destroys old items at once with no running Clip process, so the second case won't be in the game. But the Reload process obviously runs Render process right after new items are fetched from the datasource, and this way Change detector does its job. This is the only case that should be responsible for data binding during reload. But as I noticed it happens asynchronously, which may potentially cause the issue. I believe this PR had improved the situation.

As you might see the code had been merged and released as v1.3.0. If this doesn't solve your issue I would ask you to go back to the issue https://github.com/dhilt/ngx-ui-scroll/issues/120 and follow the recommendations from my first comment related to dev logs and repro.